app/controllers/concerns/eventable.rb

```diff
@@ -0,0 +1,30 @@
+module Eventable
+  extend ActiveSupport::Concern
+
+  def supplier_id
+    # NB: not all events will have a supplier_id so this could well be nil
+    @supplier_id ||= current_user.owner&.id
+  end
+
+  def process_event(moves, event_name, event_params)
```

Comment:
This all came from move events controller - I needed to add a `move` param so that it can be called for each move within an allocation. To make this nicer to use this supports passing either a single move or a move relationship - hence the array flattening.

Reply:
👍